src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,64 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String OPERATION_SUPPLY = "supply";
+    private static final String OPERATION_BUY = "buy";
+    private static final String SEPARATOR = ",";
+    private static final int OPERATION_INDEX = 0;
+    private static final int AMOUNT_INDEX = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] allLines = getAllLines(fromFileName);
+        int[] result = processDataInFile(allLines);
+        writeDataToFile(toFileName, result);
+    }
+
+    private String[] getAllLines(String fromFileName) {
+        if (fromFileName == null) {
+            throw new IllegalArgumentException("File name cannot be null " + fromFileName);
```

Comment:
it will print
"File name cannot be null null"
```suggestion
            throw new IllegalArgumentException("File name cannot be null!");
```